Formula/rustfmt.rb

```diff
@@ -20,17 +20,43 @@ class Rustfmt < Formula
 
   depends_on "rustup-init" => :build
   depends_on "rust" => :test
+  uses_from_macos "zlib"
+
+  on_linux do
+    depends_on "patchelf" => :build
+  end
 
   def install
-    system "#{Formula["rustup-init"].bin}/rustup-init", "-qy", "--no-modify-path"
+    system "rustup-init", "--profile", "minimal", "-qy", "--no-modify-path", "--default-toolchain", "none"
     ENV.prepend_path "PATH", HOMEBREW_CACHE/"cargo_cache/bin"
-    # we are using nightly because rustfmt requires nightly in order to build from source
-    # pinning to nightly-2021-11-08 to avoid inconstency
-    nightly_version = "nightly-2021-11-08"
-    components = %w[rust-src rustc-dev llvm-tools-preview]
-    system "rustup", "toolchain", "install", nightly_version
-    system "rustup", "component", "add", *components, "--toolchain", nightly_version
+
+    ENV["CFG_RELEASE_CHANNEL"] = "stable"
     system "cargo", "install", *std_cargo_args
+
+    # See https://github.com/NixOS/nixpkgs/blob/6cee3b5893090b0f5f0a06b4cf42ca4e60e5d222/pkgs/development/compilers/rust/rustfmt.nix#L18-L27
+    rust_libs = ["librustc_driver", "libstd"]
+    rust_libs << "libLLVM" if OS.linux?
+    lib_suffix = OS.mac? ? "dylib" : "so"
+    rust_libs.each do |lib|
+      (libexec/"lib").install buildpath.glob(".brew_home/.rustup/toolchains/*/lib/#{lib}-*.#{lib_suffix}")
```

Comment:
```suggestion
    rust_libs.each do |lib|
      (libexec/"lib").install buildpath.glob(".brew_home/.rustup/toolchains/*/lib/#{shared_library("#{lib}-*")}")
```
We could probably just install these straight into `libexec` directly, I think.